app/models/shipit/deploy_spec/npm_discovery.rb

```diff
@@ -40,6 +40,13 @@ def public?
         JSON.parse(file.read)['private'].blank?
       end
 
+      def dist_tag(version)
+        # All pre-release tags are treated as "next"
+        # e.g. 1.0.0-beta.1 is installable using `yarn add package@next``
+        return 'next' if version.include? '-'
```

Comment:
Can we check for an explicit `-beta` instead? 

We do some funny things in some places so this might get flagged. 
eg. https://github.com/Shopify/tinymce/blob/master/package.json#L3

Reply:
Hadn't thought of that 🤔 Great point.

Reply:
Hmm that depends. What’s the use case for -shopifyv4?

Reply:
There are some versioning concerns with TinyMCE so we had to do something custom-ish. I wouldn't be surprised if we had a couple other cases org-wide somewhere. 
cc/ @nickjs

Reply:
Thanks! Then to be safe let’s detect alpha, beta, rc, and next. Sounds good?

Reply:
In https://github.com/shopify/tinymce, we keep the package version pegged to the version of tinymce we're using, but then we provide additional meta information on top of tinymce, such as config and typings. These need their own versioning. So we publish it like so: `4.5.5-shopify1`, `4.5.5-shopify2`, etc., bumping the `shopify` version as we update the config and typings. This was the best way we could find to do it, but totally open to better options.